lua/delaytrain/init.lua

```diff
@@ -83,10 +83,18 @@ function M.enable()
             table.insert(mode_array, mode)
         end
         for _, key in ipairs(keys) do
+            -- Check that keys haven't been remapped (e.g. hjkl to dtrn)
+            local keypress = ""
+            local remapped = vim.fn.maparg(key, mode_array[1])
```

Comment:
Let's say a user has a custom normal-mode mapping for a key, but they want to keep the original mapping in visual mode. This only checks the first mode in the `keys` table to see if a mapping is available, and then overwrites that mapping for the rest of the modes as well. So the visual-mode mapping would match the normal-mode mapping.

We should loop through `mode_array` and check the remap values for each mode. If they're all the same, we can just call `vim.keymap.set` once. But if any differ, we'll want to call `vim.keymap.set` for each mode that has a different remap value.

This is admittedly a weird edge-case and not very likely to show up. But IMO it's better to prevent the user from shooting themselves in the foot and opening up another issue with their confusion.